custom_metrics/privacy.js

```diff
@@ -1,24 +1,141 @@
 
     return Array.from(document.querySelectorAll('a')).filter((a) =>
       a.innerText.match(pattern)
     ).length;
   })(),
-  iab_tcf:
-    // IAB Transparency and Consent Framework is integrated on the page.
+
+  // Consent Management Platforms
+  iab_tcf_v1:
+    // IAB Transparency and Consent Framework version 1 is integrated on the page.
     // docs v1: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework
+    // description of `__cmp`: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/CMP%20JS%20API%20v1.1%20Final.md#what-api-will-need-to-be-provided-by-the-cmp-
+    // Test site: ?
+    (typeof window.__cmp == 'function'),
+    // TODO: could improve to collect 'consent string' data (see next line), but I haven't been able to find a website that still uses v1 to test this
+    // __cmp("getConsentData", null, function(v, success) { console.log(v); })
+  iab_tcf_v2: (() => {
+    // IAB Transparency and Consent Framework version 2 is integrated on the page.
     // docs v2: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2
-    (typeof window.__cmp == 'function') |
-    (typeof window.__tcfapi == 'function'),
+    // description of `__tcfapi`: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-does-the-cmp-provide-the-api
+    // Test site: rtl.de
+    let tcData;
+    if (typeof window.__tcfapi == 'function') {
+      // based on https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#gettcdata
+      __tcfapi('getTCData', 2, (result, success) => {
+        if(success) {
+          tcData = result;
+        } else {
+          tcData = {"error": "Failed to retrieve TCData"};
+        }
+      });
+    } else {
+      tcData = null;
+    }
+    return tcData;
+  })(),
+  iab_usp: (() => {
+    // IAB US Privacy User Signal Mechanism “USP API” is integrated on the page.
+    // https://github.com/InteractiveAdvertisingBureau/USPrivacy
+    // Test site: nfl.com
+    let uspData;
+    if (typeof __uspapi == 'function') {
+      // based on https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#gettcdata
+      __uspapi('getUSPData', 1, (result, success) => {
+        if(success) {
+          uspData = result;
+        } else {
+          uspData = {"error": "Failed to retrieve USPData"};
+        }
+      });
+    } else {
+      uspData = null;
+    }
+    return uspData;
+  })(),
+
+  // Ads Transparency Spotlight Data Disclosure schema
+  // Check `meta` tag cf. https://github.com/Ads-Transparency-Spotlight/documentation/blob/main/implement.md
+  // Only for top frame; can't access child frames (same-origin policy)
```

Comment:
@VictorLeP In what case do we need it? The extension itself also looks in the website's DOM.

Reply:
I actually rather expect the frames that contain the ads to provide the transparency metadata. As far as I can tell, the extension traverses all frames (`fetch_metadata_functions.js` in the unpacked extension).